### PR TITLE
Add elasticseach shield role module

### PR DESCRIPTION
**Pull request by barryib**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

elasticsearch_shield_role
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### SUMMARY

This module manage the Elasticsearch shield role in esnative (for now) realm.

For more information refer to shield documentation.

See #2528 for shield user management.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @barryib for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: database/misc/elasticsearch_shield_role.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    state: present
+
+# Delete a role from your cluster
+- elasticsearch_shield_role:
+    name: my_kibana_user
+    state: absent
+'''
+
+RETURN = '''
+changed:
```
Comment (resmo) [inline database/misc/elasticsearch_shield_role.py]: please removed `changed` as it is a common return.

Comment (barryib) [inline database/misc/elasticsearch_shield_role.py]: Hello @resmo, it's removed. see 24e06f1

File: database/misc/elasticsearch_shield_role.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    shield_role = ShieldNativeRole(module, elasticsearch_api)
+
+    present, role = shield_role.get_role(name)
+
+    if state == 'present':
+        if not cluster or not indices:
+            module.fail_json(msg="cluster and indices are required to add role")
+
+        if present:
+            if update_role == 'always':
```
Comment (resmo) [inline database/misc/elasticsearch_shield_role.py]: It seems you always update the role no matter if the existing data is different. right? Is there a way to just compare the existing data and only update the role if required?

Comment (barryib) [inline database/misc/elasticsearch_shield_role.py]: @resmo Right. I was letting users the choice to always update the role or not. But I think that your idea is better. So I removed the `update_role` option an I update role only when something changed. See 489bfd1

Comment (resmo) [inline database/misc/elasticsearch_shield_role.py]: @barryib that would be much better, as ansible is meant to be like "just do the work if needed" and "make sure it look like this"

**Comment by gregdek** [comment]
Thanks @barryib for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by mattclay** [comment]
Tests on shippable failed because this PR branch was created before shippable was enabled. If you rebase the branch the tests will pass, although in this case the tests will just no-op since there are no tests for this module.

**Comment by barryib** [comment]
ready_for_review

**Comment by barryib** [comment]
@mattclay thanks for your help.

**Comment by gregdek** [comment]
Thanks @barryib for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**